src/test/java/fi/otavanopisto/kuntaapi/test/RestfulPtvServiceMocker.java

```diff
@@ -0,0 +1,264 @@
+  }
+  
+  @Override
+  public void startMock() {
+    super.startMock();
+    serviceMocker.start();
+  }
+  
+  @Override
+  public void endMock() {
+    serviceMocker.stop();
+    super.endMock();
+  }
+  
+  /**
+   * Mocks resources
+   * 
+   * @param ids resource ids
+   * @return mocker mocker
+   */
+  public RestfulPtvServiceMocker mockServices(String... ids) {
+    try {
+      for (String id : ids) {
+        if (!serviceMocker.isMocked(id)) {
+          mockService(readServiceFromJSONFile(String.format("services/%s.json", id)));
+          
+          ResourceMocker<String, ElectronicChannel> electronicChannelMocker = new ResourceMocker<>();
+          electronicChannelMocker.addStatusList(MockedResourceStatus.OK, urlPathEqualTo(String.format(ELECTRONIC_CHANNELS_PATH, id)));
+          serviceMocker.addSubMocker(id, electronicChannelMocker);
+          
+          ResourceMocker<String, PhoneChannel> phoneChannelMocker = new ResourceMocker<>();
+          phoneChannelMocker.addStatusList(MockedResourceStatus.OK, urlPathEqualTo(String.format(PHONE_CHANNELS_PATH, id)));
+          serviceMocker.addSubMocker(id, phoneChannelMocker);
+
+          ResourceMocker<String, PrintableFormChannel> printableFormChannelMocker = new ResourceMocker<>();
+          printableFormChannelMocker.addStatusList(MockedResourceStatus.OK, urlPathEqualTo(String.format(PRINTABLE_FORM_CHANNELS_PATH, id)));
+          serviceMocker.addSubMocker(id, printableFormChannelMocker);
+
+          ResourceMocker<String, PrintableFormChannel> locationChannelMocker = new ResourceMocker<>();
+          locationChannelMocker.addStatusList(MockedResourceStatus.OK, urlPathEqualTo(String.format(LOCATION_CHANNELS_PATH, id)));
+          serviceMocker.addSubMocker(id, locationChannelMocker);
+
+          ResourceMocker<String, WebPageChannel> webPageChannelMocker = new ResourceMocker<>();
+          webPageChannelMocker.addStatusList(MockedResourceStatus.OK, urlPathEqualTo(String.format(WEBPAGE_CHANNELS_PATH, id)));
+          serviceMocker.addSubMocker(id, webPageChannelMocker);
+        } else {
+          serviceMocker.setStatus(id, MockedResourceStatus.OK);
+        }
+      }
+    } catch (JsonProcessingException e) {
+      fail(e.getMessage());
+    }
+
+    return this;
+  }
+  
+  /**
+   * Mocks electronic service
+   * 
+   * @param ids page ids
+   * @return mocker
+   */
+  public RestfulPtvServiceMocker mockElectronicServiceChannels(String serviceId, String... ids) {
+    for (String id : ids) {
+      @SuppressWarnings("unchecked")
+      ResourceMocker<String, ElectronicChannel> channelMocker = (ResourceMocker<String, ElectronicChannel>) serviceMocker.getSubMocker(serviceId, 0);
+      if (!channelMocker.isMocked(id)) {
+        channelMocker.add(id, readElectronicChannelFromJSONFile(String.format("electronicservicechannels/%s.json", id)), urlPathEqualTo(String.format("%s/%s", String.format(ELECTRONIC_CHANNELS_PATH, serviceId), id)));
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "%s/%s" 6 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)